pkg/daemon/ceph/client/osd.go

```diff
@@ -220,6 +221,45 @@ func GetOSDUsage(context *clusterd.Context, clusterInfo *ClusterInfo) (*OSDUsage
 	return &osdUsage, nil
 }
 
+func convertKibibytesToTebibytes(kib string) (float64, error) {
+	kibFloat, err := strconv.ParseFloat(kib, 64)
+	if err != nil {
+		return float64(0), errors.Wrap(err, "failed to convert string to float")
+	}
+	return kibFloat / float64(1024*1024*1024), nil
+}
+
+func ResizeOsdCrushWeight(actualOSD OSDNodeUsage, ctx *clusterd.Context, clusterInfo *ClusterInfo) (bool, error) {
+	crushWeight, err := strconv.ParseFloat(actualOSD.CrushWeight.String(), 64)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed converting string to float for osd.%d crush weight %q", actualOSD.ID, actualOSD.CrushWeight.String())
+	}
+	// actualOSD.KB is in KiB units
+	desiredOSDTiB, err := convertKibibytesToTebibytes(actualOSD.KB.String())
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to convert KiB to TiB for osd.%d crush weight %q", actualOSD.ID, actualOSD.KB.String())
+	}
+
+	// do not reweight if the desired crush weight is less than equal or there percentage resize is less than 1 percent
+	if desiredOSDTiB <= crushWeight {
+		logger.Debugf("desired osd size is  %f is less then current osd size %f for osd.%d, not resizing the crush weights", desiredOSDTiB, crushWeight, actualOSD.ID)
+		return false, nil
+	} else if math.Abs(((desiredOSDTiB - crushWeight) / crushWeight)) <= 0.01 {
```

Comment:
Is there a unit test for these conditions?
1. If desired < current, skip the update
2. If the desired is changed less than 1%, skip the update